#Title: Differences in gene expressions between fetus and adults brains

## Summary
In this genomic data science project, we are evaluating differences in gene expression between fetal and adult brains. Transcriptome sequencing data (known as RNA-seq) for this genomic data science project was sequenced on an Illumina platform from human post-mortem brains. The raw sequencing data and meta-data related to each brain sample has been downloaded from a public database. The data will be aligned, quality control performed on the alignments, and gene count levels of expression will be determined. With the aid of exploratory and fitting statistical models,  patterns of gene expression will be identified between fetus and adult brains. Findings from this genomic data science project will provide insights in to genes differentially expressed between fetus and adult brains, and will be an example of reproducible research in genomic sciences.

### Sites to access the 1) original article, 2)  RNA-seq data, 3) phenotype meta-data, and 4) how to get the data 4

1. Find in this link the related publication: http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4281298/

2. Find in this link the RNA-seq data: https://www.dropbox.com/s/m0qgo3mo1jiiyyp/list-of-possible-samples-to-use.pdf?dl=0

3. Find in this link the phenotype meta-data for the samples: http://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA245228

4. Find in this link instructions on how to get the data: https://www.dropbox.com/s/0fnu9a0pguzjqrr/how-to-get-the-data-from-SRA__by-Coursera.pdf?dl=0

### Steps to be performed

1. Download the RNA-seq data and phenotype meta-data
2. RNA-seq sequences alignment (through **The Galaxy Project, https://usegalaxy.org**)
3. QC of alignments
4. Calculate gene expressions based on gene count levels
5. Perform exploratory analysis to refine model to be built
6. Built model to ID genes differentially expressed between fetal and adult brains
7. Identify possible biological patterns
8. Document the whole process through a brief report

Files, including codes and other relevant to the project, will be shared on this repo on a continuous bases. 


